src/aiodynamo/client.py

```diff
@@ -583,38 +625,79 @@ def table(self, name: TableName) -> Table:
-            expression_attribute_values,
-        ) = update_expression.encode()
-
-        if not update_expression:
+        expression = update_expression.encode(params)
+        if not expression:
             raise EmptyItem()
 
-        builder = ConditionExpressionBuilder()
+        payload = {
+            "TableName": table,
+            "Key": py2dy(key),
+            "UpdateExpression": expression,
+            "ReturnValues": return_values.value,
+        }
         if condition:
-            condition_expression, ean, eav = builder.build_expression(condition)
-            expression_attribute_names.update(ean)
-            expression_attribute_values.update(eav)
-        else:
-            condition_expression = None
-        resp = await self.core.update_item(
-            **clean(
-                TableName=table,
-                Key=py2dy(key),
-                UpdateExpression=update_expression,
-                ExpressionAttributeNames=expression_attribute_names,
-                ExpressionAttributeValues=py2dy(expression_attribute_values),
-                ConditionExpression=condition_expression,
-                ReturnValues=return_values.value,
-            )
-        )
+            payload["ConditionExpression"] = condition.encode(params)
+
+        payload["ExpressionAttributeNames"] = params.get_expression_names()
+        payload["ExpressionAttributeValues"] = params.get_expression_values()
+
+        resp = await self.send_request(action="UpdateItem", payload=payload)
+
         if "Attributes" in resp:
             return dy2py(resp["Attributes"], self.numeric_type)
-
         else:
             return None
+
+    async def send_request(
+        self, *, action: str, payload: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        key = await self.credentials.get_key(self.http)
+        request = signed_dynamo_request(
+            key=key,
+            payload=payload,
+            action=action,
+            region=self.region,
+            endpoint=self.endpoint,
+        )
+        attempt = 0
+        while attempt < self.throttle_config.max_attempts:
+            try:
+                return await self.http.post(
+                    url=request.url, headers=request.headers, body=request.body
+                )
+            except Throttled:
+                await asyncio.sleep(self.throttle_config.delay_func(attempt))
+                attempt += 1
+        raise Throttled()
+
+    async def _fast_depaginate(
+        self, action: str, payload: Dict[str, Any]
+    ) -> AsyncIterator[Dict[str, Any]]:
+        """
+        Internal API to depaginate the results from query/scan/count.
+        Don't call this directly, use .query, .scan or .count instead.
+        """
+        task = asyncio.create_task(self.send_request(action=action, payload=payload))
+        try:
+            while task:
+                result = await task
+                try:
+                    # TODO: Limit!
```

Comment:
is the comment perhaps no longer relevant because limit is enforced by higher level?